homeassistant/components/sensor/haveibeenpwned.py

```diff
@@ -0,0 +1,152 @@
+"""
+Support for haveibeenpwned (email breaches) sensor.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.haveibeenpwned/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+import requests
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (STATE_UNKNOWN, CONF_EMAIL)
+from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+DATE_STR_FORMAT = "%Y-%m-%d %H:%M:%S"
+USER_AGENT = "Home Assistant HaveIBeenPwned Sensor Component"
+
+SCAN_INTERVAL = 5
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_EMAIL): vol.All(cv.ensure_list, [cv.string]),
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the RESTful sensor."""
+    emails = config.get(CONF_EMAIL)
+
+    data = HaveIBeenPwnedData(emails)
+    data.update = Throttle(timedelta(seconds=SCAN_INTERVAL-1))(data.update)
```

Comment:
Why is this? Better to just request the info for each email address at startup and throttle the update as normal